src/DynamoCore/Scheduler/DynamoSchedulerInternals.cs

```diff
@@ -145,6 +158,7 @@ private void ReprioritizeTasksInQueue()
 
         private void ProcessTaskInternal(AsyncTask asyncTask)
         {
+            HasTaskInProgress = true;
```

Comment:
@BogdanZavu I saw that previously this was marked volatile.  I've never seen that modifier used in c#, only c (when doing embedded micro programming) - I found the following info:
https://stackoverflow.com/questions/72275/when-should-the-volatile-keyword-be-used-in-c

now, I see that the there is no concurrency primitives used, locks etc. Do we need to consider wrapping this accessor in a lock, or will the scheduler threadbe the only thread to interact with it?

Reply:
I kind of disagree with that statement, especially for a simple bool var case like ours but let's not spend the weekend debating over it :) I will use lock